### DISCUSSION TITLE
Adjust vertical/horizontal image scan in page

### DISCUSSION THREAD

**Discussion by VadeInPace** [ImageMagick/ImageMagick#6335] [Help] [answered]
### ImageMagick version

7.1.1-4

### Operating system, version and so on

Opensuse Leap 15.4 MATE desktop 1.26.0

### Description

Hello there,
Last month I posted here and I still am very happy by the script snibgo wrote for me. Thanks again.
Really could not write it even after months of learning.
Today I need another one, I hope easier.
I scan some old music sheet. Sometime the music stave are not placed horizontally/vertically in the page scan, see attached sample.
Is is possible to write a script that take the lines of the stave as reference to adjust the horizontal sheet edge and centre it in the page.
This sample is the worse example I could have.
![MusicHVtest](https://github.com/ImageMagick/ImageMagick/assets/49396694/c0a0849b-8f3f-45c3-b230-8d95cc2da152)

Thanks for your help

**Comment by fmw42** [comment]
Try

`magick music.png -deskew 40% x.png`

![x](https://github.com/ImageMagick/ImageMagick/assets/13458218/06d35ff7-8aac-4673-9f73-ca647e9080fd)

**Reply by VadeInPace** [reply]
This is really magic!
But why the size of the file change from ~88 kB to ~554 kB? This is a huge increase, it could be crippling.
Is it possible to reduce the size?
Thanks

**Comment by snibgo** [comment, marked as answer]
As fmw42 says, but I suggest adding "`+repage`" after the deskew, to remove canvas offsets.

In this example, we get a better result if we isolate the first stave and deskew that. Then we find what angle was used, and use that angle to rotate the original image. Windows BAT syntax, where `musicSkewed.png` is your supplied input example:
```
magick ^
  musicSkewed.png +write mpr:INP ^
  -crop 2270x392+143+414 +repage ^
  -deskew 40% +repage ^
  -set option:ANG %%[deskew:angle] ^
  +delete ^
  mpr:INP ^
  -rotate %%[ANG] +repage ^
  x1.png
```

The file size has increased because your input is binary, black and white only. Rotating it introduces greys. I suggest you ensure the input is greyscale, and only make it black and white as the final operation after any rotations, stitching and whatever else.

**Reply by fmw42** [reply]
Sorry, I forgot the +repage.  Thanks, snibgo.

**Reply by VadeInPace** [reply]
Many thanks, working
Changed to Linux Bash and added change to B&W
```
#!/bin/bash

magick \
  musicgr.png +write mpr:INP \
  -crop 2270x392+143+414 +repage \
  -deskew 40% +repage \
  -set option:ANG %[deskew:angle] \
  +delete \
  mpr:INP \
  -rotate %[ANG] +repage \
  -depth 1 \
  outgrbw.png
```
Files size are,
B&W input 86.9MB become B&W output 86.7MB
greyscale input 2.1MB become B&W output 57.2KB
I'll scan all as greyscale and transform at the end of the process.

Last question (I think)
What make the length and width of the output page?

**Comment by snibgo** [comment]
> What make the length and width of the output page?

I don't understand the question.

When a rectangle is rotated through a small angle, it takes up more space. So the output width and height is more than the input width and height. If you need to know the output dimensions, you can do this:
```
-format "%w %h" +write info:
```

**Reply by VadeInPace** [reply]
I didn't understand your answer at first because I was viewing only the printed  (black) space. In fact the full page is rotated!
I get it.
In fact the "text" in the output page is narrower the in the input.
Now, to get all the same size pages, I will try the commands you gave me in another thread
```
for file in *.png
do
echo "$file"
magick \
  "$file" +write mpr:INP \
  -crop 2270x392+143+414 +repage \
  -deskew 40% +repage \
  -set option:ANG %[deskew:angle] \
  +delete \
  mpr:INP \
  -rotate %[ANG] +repage \
  -background White -gravity Center -extent 2610x3710 \
  -gravity Center -crop 2610x3710+0+0 +repage \
  -depth 1 \
  "$outdir/$file"
done
```
I'll have to adjust the size depending of the scan resolution.
I'll play with that, is it right?

**Comment by snibgo** [comment]
The crop I showed for the first stave `-crop 2270x392+143+414` may not be correct for all your `*.png` files.
